sympy/solvers/tests/test_solveset.py

```diff
@@ -1961,3 +1867,119 @@ def test_solve_logarithm():
+    eq = 3*log(a**(3*x + 5)) + b*log(a**(3*x + 5)) + a**(3*x + 5)
+    assert solveset_real(eq, x) == FiniteSet(
+        (-log(a**5) - LambertW(S(1)/(b + 3)))/(3*log(a)))
+
+    p = symbols('p', positive=True)
+    eq = 3*log(p**(3*x + 5)) + p**(3*x + 5)
+    assert solveset(eq, x) == FiniteSet(log(3**(S(1)/3)*LambertW(S(1)/3)**(S(1)/3)/p**(S(5)/3))/log(p),
+    log(-3**(S(1)/3)*LambertW(S(1)/3)**(S(1)/3)/(2*p**(S(5)/3)) - 3**(S(5)/6)*I*LambertW(S(1)/3)**(S(1)/3)/(2*p**(S(5)/3)))/log(p),
+    log(-3**(S(1)/3)*LambertW(S(1)/3)**(S(1)/3)/(2*p**(S(5)/3)) + 3**(S(5)/6)*I*LambertW(S(1)/3)**(S(1)/3)/(2*p**(S(5)/3)))/log(p))
+    assert solveset_real(eq, x) == FiniteSet(-S(5)/3 - LambertW(S(1)/3)/(3*log(p)))
+
+    assert solveset_real(2*x + 5 + log(3*x - 2), x) == \
+        FiniteSet(Rational(2, 3) + LambertW(2*exp(-Rational(19, 3))/3)/2)
+
+    assert solveset_real(3*x + log(4*x), x) == \
+        FiniteSet(LambertW(Rational(3, 4))/3)
+
+    assert solveset_real(a/x + exp(x/2), x) == \
+        FiniteSet(2*LambertW(-a/2)) - FiniteSet(0)
+    assert solveset_real(x*exp(x) - 1, x) == FiniteSet(LambertW(1))
+    eq = (x*exp(x) - 3).subs(x, x*exp(x))
+    assert solveset_real(eq, x) == \
+        FiniteSet(LambertW(3*exp(-LambertW(3))))
+
+    assert solveset_real(x*log(x) + 3*x + 1, x) == \
+        Intersection(Interval(0, oo), FiniteSet(exp(-3 + LambertW(-exp(3)))))
+    # issue 4271
+    assert solveset_real((a/x + exp(x/2)).diff(x, 2), x) == Intersection(
+        Interval(0, oo), FiniteSet(6*LambertW((-a)**(S(1)/3)/3))) - FiniteSet(0)
+    assert solveset_real(x**3 - 3**x, x) == FiniteSet(
+        -3*LambertW(-log(3)/3)/log(3), -3*LambertW(-log(3)/3, -1)/log(3))
+    assert solveset_real(x**2 - 2**x, x) == solveset_real(-x**2 + 2**x, x)
+
+    assert solveset_real((a/x + exp(x/2)).diff(x), x) == \
+        Intersection(Interval(0, oo),
+            FiniteSet(4*LambertW(sqrt(2)*sqrt(a)/4))) - FiniteSet(0)
+    a = -1
+    assert solveset_real((a/x + exp(x/2)).diff(x), x) == S.EmptySet
+    a = 1
+    assert solveset_real((a/x + exp(x/2)).diff(x), x) == FiniteSet(
+        4*LambertW(sqrt(2)/4))
+
+
+@XFAIL
+def test_other_solve_lambert_():
+    a = Symbol('a')
+
+    assert solveset_real(-a*x + 2*x*log(x), x) == \
+        Union(FiniteSet(0), Intersection(S.Reals, FiniteSet(exp(a/2))))
+    assert solveset_real(log(log(x - 3)) + log(x-3), x) == FiniteSet(
+        exp(LambertW(1)) + 3)
+
+
+def test_solve_bivariate():
+
+    assert solveset_real((x**2 - 2*x + 1).subs(x, log(x) + 3*x), x) == \
+        FiniteSet(LambertW(3*S.Exp1)/3)
+    assert solveset_real((x**2 - 2*x + 1).subs(x, (log(x) + 3*x)**2 - 1), x) == \
+        FiniteSet(LambertW(3*exp(-sqrt(2)))/3, LambertW(3*exp(sqrt(2)))/3)
+    assert solveset_real((x**2 - 2*x - 2).subs(x, log(x) + 3*x), x) == \
+        FiniteSet(LambertW(3*exp(1 + sqrt(3)))/3, LambertW(3*exp(-sqrt(3) + 1))/3)
+
+
+@XFAIL
+def test_other_solve_bivariate():
+    eq = 2*(3*x + 4)**5 - 6*7**(3*x + 9)
```

Comment:
This test can be passed if we take idea from here 
https://github.com/sympy/sympy/blob/faad12d1307099aff62b559a84df40a999b8219c/sympy/solvers/solvers.py#L2757-L2773

Although I m on it . I just wanted to let you know.